### PR TITLE
[PM-5277] Move SyncService State

### PR DESCRIPTION
**Pull request by justindbaur**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-5277

## 📔 Objective

Moves the `lastSync` state from state service to state providers. We also got to delete the web state service 🥳.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

**Review by quexten — DISMISSED** [review]
Looks good for the auth-owned code. Great work!

File: libs/common/src/state-migrations/migrate.ts [hunk 1]
```diff
@@ -138,7 +140,9 @@ export function createMigrationBuilder() {
     .with(PasswordOptionsMigrator, 62, 63)
     .with(GeneratorHistoryMigrator, 63, 64)
     .with(ForwarderOptionsMigrator, 64, 65)
-    .with(MoveFinalDesktopSettingsMigrator, 65, CURRENT_VERSION);
+    .with(MoveFinalDesktopSettingsMigrator, 65, 66)
+    .with(RemoveUnassignedItemsBannerDismissed, 66, 67)
```
Comment (dani-garcia) [inline libs/common/src/state-migrations/migrate.ts:144]: Was adding this migration here forgotten when the migration file was created? Would be nice if we could have a lint to remind us that the file was unused.

**Review by dani-garcia — APPROVED** [review]
LGTM!

**Review by quexten — APPROVED** [review]

File: libs/common/src/state-migrations/migrate.ts (hunk 1 above)
Comment (justindbaur) [reply libs/common/src/state-migrations/migrate.ts:144]: Yep :/, from my quick looking I don't see a built in rule for that unfortunately but that would be really nice.

**Pull request merged**